include/deal.II/lac/sparse_matrix_ez.h

```diff
@@ -683,6 +683,17 @@ class SparseMatrixEZ : public Subscriptor
   const_iterator begin (const size_type r) const;
 
   /**
+   * Iterator on diagonal entry of row @p r.
+   *
+   * @warning If the current row has no diagonal element, then the returned
+   * iterator will be same as end(r), i.e., an iterator points to first element
+   * of next row or same place behind end of the entire matrix.
+   * So, do not dereference or query information to the returned iterator blindly
+   * unless you can make sure your matrix do have a diagonal element in row @p r.
+   */
+  const_iterator diagonal (const size_type r) const;
```

Comment:
I want to get a iterator on diagonal element because I want to do loop begin or end on diagonal,
say, during ILU decomposition.

This function is not designed to get value of a diagonal element, which can be done easily via
```
el(i,i)
````